crates/ruma-common/src/events/ignored_user_list.rs

```diff
@@ -14,103 +16,32 @@ use crate::OwnedUserId;
-    pub fn new(ignored_users: Vec<OwnedUserId>) -> Self {
+    /// Creates a new `IgnoredUserListEventContent` from the given map of ignored user.
+    pub fn new(ignored_users: BTreeMap<OwnedUserId, IgnoredUser>) -> Self {
         Self { ignored_users }
     }
-}
 
-mod vec_as_map_of_empty {
-    use std::{fmt, marker::PhantomData};
-
-    use serde::{
-        de::{self, Deserialize, Deserializer},
-        ser::{SerializeMap, Serializer},
-        Serialize,
-    };
-
-    /// Serialize the given `Vec<T>` as a map of `T => Empty`.
-    #[allow(clippy::ptr_arg)]
-    pub fn serialize<S, T>(vec: &Vec<T>, serializer: S) -> Result<S::Ok, S::Error>
-    where
-        S: Serializer,
-        T: Serialize + Eq + Ord,
-    {
-        let mut map = serializer.serialize_map(Some(vec.len()))?;
-        for item in vec {
-            map.serialize_entry(item, &Empty {})?;
-        }
-        map.end()
-    }
-
-    /// Deserialize an object and return the keys as a `Vec<T>`.
-    pub fn deserialize<'de, D, T>(deserializer: D) -> Result<Vec<T>, D::Error>
-    where
-        D: Deserializer<'de>,
-        T: Deserialize<'de> + Eq + Ord,
-    {
-        struct MapOfEmptyVisitor<T>(PhantomData<T>);
-        impl<'de, T> de::Visitor<'de> for MapOfEmptyVisitor<T>
-        where
-            T: Deserialize<'de>,
-        {
-            type Value = Vec<T>;
-
-            fn expecting(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-                write!(f, "an object/map")
-            }
-
-            fn visit_map<A>(self, mut map: A) -> Result<Self::Value, A::Error>
-            where
-                A: de::MapAccess<'de>,
-            {
-                let mut items = Vec::with_capacity(map.size_hint().unwrap_or(0));
-                while let Some((item, _)) = map.next_entry::<T, Empty>()? {
-                    items.push(item);
-                }
-                Ok(items)
-            }
-        }
-
-        deserializer.deserialize_map(MapOfEmptyVisitor(PhantomData))
-    }
-
-    #[derive(Clone, Debug, Serialize)]
-    struct Empty {}
-
-    impl<'de> Deserialize<'de> for Empty {
-        fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
-        where
-            D: Deserializer<'de>,
-        {
-            struct EmptyMapVisitor;
-
-            impl<'de> de::Visitor<'de> for EmptyMapVisitor {
-                type Value = Empty;
-
-                fn expecting(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-                    write!(f, "an object/map")
-                }
-
-                fn visit_map<A>(self, _map: A) -> Result<Self::Value, A::Error>
-                where
-                    A: de::MapAccess<'de>,
-                {
-                    Ok(Empty {})
-                }
-            }
-
-            deserializer.deserialize_map(EmptyMapVisitor)
-        }
+    /// Creates a new `IgnoredUserListEventContent` from the given list of users.
+    pub fn users(ignored_users: Vec<OwnedUserId>) -> Self {
```

Comment:
I think we should generalize this to any `impl IntoIterator<Item = OwnedUserId>`.